### PR TITLE
feat(apib): upgraded nunjucks and added more tests for serialiser

### PR DESCRIPTION
**Pull request by pksunkara**
Look in the comments for description

### CODE REVIEW

File: packages/fury/lib/fury.js [hunk 1]
```diff
@@ -268,7 +268,10 @@ class Fury {
     const promise = adapter.serialize({ api, namespace: this.minim, mediaType });
 
     if (done) {
-      promise.then(result => done(null, result), done);
+      promise
```
Comment (kylef) [inline packages/fury/lib/fury.js]: Why did you change this?

File: packages/fury-adapter-apib-serializer/template.nunjucks [hunk 2]
```diff
@@ -77,9 +78,9 @@ FORMAT: 1A
 
 {% macro renderTransition(transition, href) %}
 {% if transition.title and transition.title.toValue() %}
-#### {{ transition.title.toValue() }} [{{ transition.method.toValue() }}{% if href %} {{ href }}{% elif transition.computedHref and transition.computedHref.toValue() %} {{ transition.computedHref.toValue() }}{% endif %}]
+#### {{ transition.title.toValue() }} [{% if transition.method %}{{ transition.method.toValue() }}{% endif %}{% if href %} {{ href }}{% elif transition.computedHref and transition.computedHref.toValue() %} {{ transition.computedHref.toValue() }}{% endif %}]
```
Comment (pksunkara) [inline packages/fury-adapter-apib-serializer/template.nunjucks]: When an action is missing a response,

```apib
FORMAT: 1A

# API name

## Group Users

### User [/users]

#### List all users [GET]

Get the list of all the users
```

The output tree is

```json
{
  "element": "parseResult",
  "content": [
    {
      "element": "category",
      "meta": {
        "classes": {
          "element": "array",
          "content": [
            {
              "element": "string",
              "content": "api"
            }
          ]
        },
        "title": {
          "element": "string",
          "content": "API name"
        }
      },
      "attributes": {
        "metadata": {
          "element": "array",
          "content": [
            {
              "element": "member",
              "meta": {
                "classes": {
                  "element": "array",
                  "content": [
                    {
                      "element": "string",
                      "content": "user"
                    }
                  ]
                }
              },
              "content": {
                "key": {
                  "element": "string",
                  "content": "FORMAT"
                },
                "value": {
                  "element": "string",
                  "content": "1A"
                }
              }
            }
          ]
        }
      },
      "content": [
        {
          "element": "category",
          "meta": {
            "classes": {
              "element": "array",
              "content": [
                {
                  "element": "string",
                  "content": "resourceGroup"
                }
              ]
            },
            "title": {
              "element": "string",
              "content": "Users"
            }
          },
          "content": [
            {
              "element": "resource",
              "meta": {
                "title": {
                  "element": "string",
                  "content": "User"
                }
              },
              "attributes": {
                "href": {
                  "element": "string",
                  "content": "/users"
                }
              },
              "content": [
                {
                  "element": "transition",
                  "meta": {
                    "title": {
                      "element": "string",
                      "content": "List all users"
                    }
                  },
                  "content": [
                    {
                      "element": "copy",
                      "content": "Get the list of all the users"
                    }
                  ]
                }
              ]
            }
          ]
        }
      ]
    }
  ]
}

```

And when you serialize this, the current master crashes.

File: packages/fury-adapter-apib-serializer/test/fixtures/basic.apib [hunk 3]
```diff
@@ -2,12 +2,8 @@ FORMAT: 1A
 
 # My API
 
-## Group A
-
 ### /message
 
 #### GET
 
-+ Response 200 (text/plain)
```
Comment (kylef) [inline packages/fury-adapter-apib-serializer/test/fixtures/basic.apib]: With this particular change, I think we don't have any test coverage which has a content-type in response payload. Could you either restore it or create a specific test with request/response with content-types?

File: packages/fury/lib/fury.js (hunk 1 above)
Comment (pksunkara) [reply packages/fury/lib/fury.js]: It was needed, otherwise there were a lot of warnings getting printed.

File: packages/fury-adapter-apib-serializer/test/fixtures/basic.apib (hunk 3 above)
Comment (pksunkara) [reply packages/fury-adapter-apib-serializer/test/fixtures/basic.apib]: `sample.apib` contains the test case.

**Comment by pksunkara** [comment]
There are lot more issues that needs to be covered and are currently failing.

File: packages/fury/lib/fury.js (hunk 1 above)
Comment (kylef) [reply packages/fury/lib/fury.js]: Can you perhaps explain the difference a bit further or give me some steps I can reproduce these warnings?

There's other cases where this syntax is used

Comment (pksunkara) [reply packages/fury/lib/fury.js]:
```
(node:10542) UnhandledPromiseRejectionWarning: AssertionError: expected 'FORMAT: 1A\n\n# My API\n\n## Group A\n\n### /message\n\n#### GET\n\n+ Response 200 (text/plain)\n\n        Hello, World!' to deeply equal 'FORMAT: 1A\n\n# My API\n\n## Group A\n\n### /message\n\n#### GET\n\n+ Response 200 (text/plain)\n\n        Hello, World'
    at /Users/pksunkara/Coding/apiary/api-elements.js/packages/fury-adapter-apib-serializer/test/adapter-test.js:46:43
    at /Users/pksunkara/Coding/apiary/api-elements.js/packages/fury/lib/fury.js:271:30
    at processTicksAndRejections (internal/process/task_queues.js:89:5)
(node:10542) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:10542) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
